src/utils/wallet.js

```diff
@@ -211,6 +214,34 @@ const wallet = {
     }
   },
 
+  /**
+   * The wallet needs each token metadata to show information correctly
+   * So we fetch the tokens metadata and store on redux
+   *
+   * @param {Array} tokens Array of token uids
+   * @param {String} network Network name
+   *
+   * @memberof Wallet
+   * @inner
+   **/
+  async fetchTokensMetadata(tokens, network) {
+    const metadatas = {};
+
+    for (const token of tokens) {
+      try {
+        const response = await metadataApi.getDag(token, network);
+        if (response.data && token in response.data) {
+          const tokenMeta = response.data[token];
+          metadatas[token] = tokenMeta;
+        }
+      } catch (e) {
+        // No need to do anything, the metadata for this token was not found
```

Comment:
What if we get a 500 or 429 return code? Or the server times out?

Reply:
We could make the api return 200 with empty body instead of 404, so this catch can be only for unexpected errors

Reply:
I've refactored this part of code to have an error handling and auto retry. Can you see if it's good? https://github.com/HathorNetwork/hathor-wallet/pull/228/commits/9e150da7e7f5923df00911378430609f9345f9aa

Reply:
After a second refactor, this code moved to the hathor-wallet-lib.